core/state_transition.go

```diff
@@ -309,8 +314,15 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 			st.state.RevertToSnapshot(snap)
 			// Even though we revert the state changes, always increment the nonce for the next deposit transaction
 			st.state.SetNonce(st.msg.From(), st.state.GetNonce(st.msg.From())+1)
+
+			// Record deposits as using all their gas (matches the gas pool)
+			// System Transactions are special & are not recorded as using any gas (anywhere)
+			gasUsed := st.msg.Gas()
+			if st.msg.IsSystemTx() {
+				gasUsed = 0
+			}
```

Comment:
This should also be done in the non-failing deposit case.

Reply:
nice catch

Reply:
fixed